### ISSUE TITLE
Add type check support for Object.create and Object.defineProperties

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#302]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The original discussion is archived at:
http://blickly.github.io/closure-compiler-issues/#455

**Comment by SunboX** [comment]
+1 :smirk:

**Comment by michaelhogg** [comment]
Is there any news regarding type-checking for objects created using `Object.create()`?

I would find this extremely helpful, for situations like this:

```js
var a = {};

/** @param {number} x */
a.myFunction = function (x) {
    window.console.log(x);
};

var b = Object.create(a);

b.myFunction('not a number');  // Should raise TypeValidator.TYPE_MISMATCH_WARNING
```

**Comment by mbrowne** [comment]
+1. This is not type-checking for me:

	/* @flow */

	Object.defineProperties({}, {
		foo: {
			get: function() {
				return '';
			}
		}
	});

I get this error:

```
$ flow
Account.js:3
  3: Object.defineProperties({}, {
     ^ .foo of Object.defineProperties. Property not found in
  4: 	foo: {
     	     ^ object literal


Found 1 error
```

**Comment by Dominator008** [comment]
@mbrowne This seems like an error log from [Flow](https://github.com/facebook/flow) :) .

**Comment by mbrowne** [comment]
Oops, you're right - please disregard.

**Comment by ILOVEPIE** [comment]
This Isn't fixed!

**Comment by lauraharker** [comment]
We mass-closed all issues tagged NTI a few months ago. (https://github.com/google/closure-compiler/wiki/%5BOBSOLETE%5D-Using-NTI-(new-type-inference))

I'll reopen this issue since it wasn't actually specific to NTI.

**Comment by EatingW** [comment]
Created Google internal issue b/120557713.
